spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -316,13 +327,146 @@ In order to track all of the denominations moving around the network of chains i
 - Each chain, locally, could elect to keep a lookup table to use short, user-friendly local denominations in state which are translated to and from the longer denominations when sending and receiving packets. 
 - Additional restrictions may be imposed on which other machines may be connected to & which channels may be established.
 
+## Extension: ICS20-2
+
+The above specifications defines `ics20-1`, which is the first version of the specification. Here we define
+ICS20-2 which 
+
+### Goals
+
+- Add support for opt-in, decentralized, pseudonymous relayer incentivization
+- Backwards compatibility with ics20-1
+- Minimal extra logic needed to maintain backwards compatibility
+- Fault-resilient: version implementation errors should not break invariants
+
+### Data Structures
+
+In order to help with backwards compatibility, we define a packet structure for ICS20-2, which is a superset of ICS20-1, such that any valid packet for one protocol, is also a valid packet for the other protocol. The only issue is that the `fee` info from a ICS20-2 packet will be ignored by a ICS20-1 handler.
+
+```typescript
+interface FungibleTokenPacketData {
+  denomination: string
+  amount: uint256
+  sender: string
+  receiver: string
+  fee: Maybe<uint256>
+}
+```
+
+Fee is defined here as some number of tokens to be sent to the address who submitted the IbcReceivePacket on the destination chain. The information of `signer` is available when submitting the packet, but discarded in the application-specific handlers for `ibc-go`. We can simply expose that information to the application to allow it to optionally take action based on who submitted the packet. Note that this means *anyone* can submit a valid packet and we do not hard code an allowed relayer nor force the token sender to select a relayer.
+
```

Comment:
Do we want to consider a relayer address to be optionally provided by the message sender to support permissioned relaying? If field is left empty (default) anyone can relay.